### PR TITLE
Add a test that verifies pods with resource limits schedule.

### PR DESCRIPTION
**Pull request by brendandburns**
@zmerlynn @quinton-hoole @dchen1107

### CODE REVIEW

**Comment by brendandburns** [comment]
Please comment, but don't merge this, I'm parallelizing getting a review, and validating the e2e passes.

**Comment by brendandburns** [comment]
Ok, this passed e2e.  ready for review and merge.

File: test/e2e/pods.go [hunk 1]
```diff
@@ -155,6 +156,42 @@ var _ = Describe("Pods", func() {
 			},
 		})
 	})
+	It("should be schedule with cpu and memory limits", func() {
+		podClient := c.Pods(api.NamespaceDefault)
```
Comment (quinton-hoole) [inline test/e2e/pods.go]: nit: Needs to run in it's own namespace using @lavalamp's Framework.  But that's not how pods.go is wired up right now, so lets defer that to a separate PR.

File: test/e2e/pods.go [hunk 2]
```diff
@@ -155,6 +156,42 @@ var _ = Describe("Pods", func() {
...
+
+		By("creating the pod")
+		name := "pod-update-" + string(util.NewUUID())
```
Comment (quinton-hoole) [inline test/e2e/pods.go]: nit: Don't need the UUID when it's running in its own namespace as per previous comment.  Defer to sep PR.

**Comment by quinton-hoole** [comment]
LGTM

File: test/e2e/pods.go (hunk 1 above)
Comment (brendandburns) [inline test/e2e/pods.go]: ack.

File: test/e2e/pods.go (hunk 2 above)
Comment (brendandburns) [inline test/e2e/pods.go]: ack.

**Comment by krousey** [comment]
@brendandburns shippable is complaining about gofmt on pods.go

**Comment by dchen1107** [comment]
LGTM thanks!

**Comment by brendandburns** [comment]
@quinton-hoole added a TODO to convert to the framework stuff
@krousey ran gofmt again, pre-commit hooks don't complain...

ptal.

Thanks
--brendan

**Comment by brendandburns** [comment]
@timothysc acknowledged, but it covers a gap that we had, and we can expand from here.

**Pull request merged**